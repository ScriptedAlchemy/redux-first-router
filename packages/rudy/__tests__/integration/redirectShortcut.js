import createTest from '../../__helpers__/createTest'

createTest('redirect to type specified as redirect route option value', {
  SECOND: {
    path: '/second',
    redirect: 'REDIRECTED',
  },
})

createTest(
  'redirect shortcut /w params passed on',
  {
    SECOND: {
      path: '/second/:param',
      redirect: 'REDIRECTED',
    },
    REDIRECTED: {
      path: '/redirected/:param',
      onComplete: () => 'redirect_complete',
    },
  },
  [{ type: 'SECOND', params: { param: 'foo' } }],
)
