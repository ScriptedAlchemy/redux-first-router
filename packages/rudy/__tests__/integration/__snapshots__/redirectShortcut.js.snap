// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`redirect shortcut /w params passed on: 1 - response 1`] = `
Object {
  "payload": "redirect_complete",
  "type": "REDIRECTED_COMPLETE",
}
`;

exports[`redirect shortcut /w params passed on: 1 - routes - NEVER_USED_PATHLESS - thunk 1`] = `0`;

exports[`redirect shortcut /w params passed on: 1 - routes - REDIRECTED - onComplete 1`] = `1`;

exports[`redirect shortcut /w params passed on: 1 - state 1`] = `
Object {
  "location": Object {
    "basename": "",
    "blocked": null,
    "direction": "forward",
    "entries": Array [
      Object {
        "basename": "",
        "hash": "",
        "location": Object {
          "key": "345678",
          "pathname": "/first",
          "scene": "",
          "search": "",
          "url": "/first",
        },
        "params": Object {},
        "query": Object {},
        "state": Object {},
        "type": "FIRST",
      },
      Object {
        "basename": "",
        "hash": "",
        "location": Object {
          "key": "345678",
          "pathname": "/redirected/foo",
          "scene": "",
          "search": "",
          "url": "/redirected/foo",
        },
        "params": Object {
          "param": "foo",
        },
        "query": Object {},
        "state": Object {},
        "type": "REDIRECTED",
      },
    ],
    "from": Object {
      "basename": "",
      "hash": "",
      "location": Object {
        "index": 1,
        "key": "345678",
        "pathname": "/second/foo",
        "scene": "",
        "search": "",
        "url": "/second/foo",
      },
      "params": Object {
        "param": "foo",
      },
      "query": Object {},
      "state": Object {},
      "type": "SECOND",
    },
    "hash": "",
    "index": 1,
    "key": "345678",
    "kind": "replace",
    "length": 2,
    "n": 1,
    "params": Object {
      "param": "foo",
    },
    "pathname": "/redirected/foo",
    "pop": false,
    "prev": Object {
      "basename": "",
      "hash": "",
      "location": Object {
        "index": 0,
        "key": "345678",
        "pathname": "/first",
        "scene": "",
        "search": "",
        "url": "/first",
      },
      "params": Object {},
      "query": Object {},
      "state": Object {},
      "type": "FIRST",
    },
    "query": Object {},
    "ready": true,
    "scene": "",
    "search": "",
    "state": Object {},
    "status": 301,
    "type": "REDIRECTED",
    "universal": false,
    "url": "/redirected/foo",
  },
  "title": "REDIRECTED_COMPLETE - \\"redirect_complete\\"",
}
`;

exports[`redirect shortcut /w params passed on: 1 - title 1`] = `"REDIRECTED"`;

exports[`redirect to type specified as redirect route option value: 1 - response 1`] = `
Object {
  "payload": "redirect_complete",
  "type": "REDIRECTED_COMPLETE",
}
`;

exports[`redirect to type specified as redirect route option value: 1 - routes - NEVER_USED_PATHLESS - thunk 1`] = `0`;

exports[`redirect to type specified as redirect route option value: 1 - routes - REDIRECTED - onComplete 1`] = `1`;

exports[`redirect to type specified as redirect route option value: 1 - state 1`] = `
Object {
  "location": Object {
    "basename": "",
    "blocked": null,
    "direction": "forward",
    "entries": Array [
      Object {
        "basename": "",
        "hash": "",
        "location": Object {
          "key": "345678",
          "pathname": "/first",
          "scene": "",
          "search": "",
          "url": "/first",
        },
        "params": Object {},
        "query": Object {},
        "state": Object {},
        "type": "FIRST",
      },
      Object {
        "basename": "",
        "hash": "",
        "location": Object {
          "key": "345678",
          "pathname": "/redirected",
          "scene": "",
          "search": "",
          "url": "/redirected",
        },
        "params": Object {},
        "query": Object {},
        "state": Object {},
        "type": "REDIRECTED",
      },
    ],
    "from": Object {
      "basename": "",
      "hash": "",
      "location": Object {
        "index": 1,
        "key": "345678",
        "pathname": "/second",
        "scene": "",
        "search": "",
        "url": "/second",
      },
      "params": Object {},
      "query": Object {},
      "state": Object {},
      "type": "SECOND",
    },
    "hash": "",
    "index": 1,
    "key": "345678",
    "kind": "replace",
    "length": 2,
    "n": 1,
    "params": Object {},
    "pathname": "/redirected",
    "pop": false,
    "prev": Object {
      "basename": "",
      "hash": "",
      "location": Object {
        "index": 0,
        "key": "345678",
        "pathname": "/first",
        "scene": "",
        "search": "",
        "url": "/first",
      },
      "params": Object {},
      "query": Object {},
      "state": Object {},
      "type": "FIRST",
    },
    "query": Object {},
    "ready": true,
    "scene": "",
    "search": "",
    "state": Object {},
    "status": 301,
    "type": "REDIRECTED",
    "universal": false,
    "url": "/redirected",
  },
  "title": "REDIRECTED_COMPLETE - \\"redirect_complete\\"",
}
`;

exports[`redirect to type specified as redirect route option value: 1 - title 1`] = `"REDIRECTED"`;
